mcs/class/test-helpers/SocketResponder.cs

```diff
@@ -91,8 +91,11 @@ private void Listen ()
 					listenSocket = tcpListener.AcceptSocket ();
 					listenSocket.Send (requestHandler (listenSocket));
 					try {
-						listenSocket.Shutdown (SocketShutdown.Receive);
+						// On Windows a Receive() is needed here before Shutdown() to consume the data some tests send.
+						listenSocket.ReceiveTimeout = 1000;
```

Comment:
Bumping the timeout to a large value seems cleaner to me IMHO if the only purpose is preventing a hang.